nifi-nar-bundles/nifi-azure-bundle/nifi-azure-reporting-task/src/main/java/org/apache/nifi/reporting/azure/loganalytics/AzureLogAnalyticsReportingTask.java

```diff
@@ -0,0 +1,326 @@
+            .description("The name of the exporting job")
+            .defaultValue("nifi_reporting_job")
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    static final PropertyDescriptor SEND_JVM_METRICS = new PropertyDescriptor.Builder()
+            .name("Send JVM-metrics")
+            .description("Send JVM-metrics in addition to the Nifi-metrics")
+            .allowableValues("true", "false")
+            .defaultValue("false")
+            .required(true)
+            .build();
+
+    private String createAuthorization(String workspaceId, String key, int contentLength, String rfc1123Date) {
+        try {
+            // Documentation: https://docs.microsoft.com/en-us/rest/api/loganalytics/create-request
+            String signature = String.format("POST\n%d\napplication/json\nx-ms-date:%s\n/api/logs", contentLength, rfc1123Date);
+            Mac mac = Mac.getInstance(HMAC_SHA256_ALG);
+            mac.init(new SecretKeySpec(DatatypeConverter.parseBase64Binary(key), HMAC_SHA256_ALG));
+            String hmac = DatatypeConverter.printBase64Binary(mac.doFinal(signature.getBytes(UTF8)));
+            return String.format("SharedKey %s:%s", workspaceId, hmac);
+        } catch (NoSuchAlgorithmException | InvalidKeyException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @OnScheduled
+    public void setup(final ConfigurationContext context) {
+       virtualMachineMetrics = JmxJvmMetrics.getInstance();
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(LOG_ANALYTICS_WORKSPACE_ID);
+        properties.add(LOG_ANALYTICS_CUSTOM_LOG_NAME);
+        properties.add(LINUX_PRIMARY_KEY);
+        properties.add(APPLICATION_ID);
+        properties.add(INSTANCE_ID);
+        properties.add(PROCESS_GROUP_IDS);
+        properties.add(JOB_NAME);
+        properties.add(SEND_JVM_METRICS);
+        return properties;
+    }
+
+    @Override
+    public void onTrigger(final ReportingContext context) {
+        final String workspaceId = context.getProperty(LOG_ANALYTICS_WORKSPACE_ID).evaluateAttributeExpressions().getValue();
+        final String linuxPrimaryKey = context.getProperty(LINUX_PRIMARY_KEY).evaluateAttributeExpressions().getValue();
+        final boolean jvmMetricsCollected = context.getProperty(SEND_JVM_METRICS).asBoolean();
+
+        String logName = context.getProperty(LOG_ANALYTICS_CUSTOM_LOG_NAME).evaluateAttributeExpressions().getValue();
+        final String instanceId = context.getProperty(INSTANCE_ID).evaluateAttributeExpressions().getValue();
+        String groupIds = context.getProperty(PROCESS_GROUP_IDS).evaluateAttributeExpressions().getValue();
+        try {
+            if(groupIds == null) {
```

Comment:
Is groupIds a valid state for the follow on condition? If not, perhaps use StringUtils.IsBlank or the like.